### ISSUE TITLE
DateTime field 

### ISSUE THREAD

**Issue by nejckorosec** [whitecube/nova-flexible-content#14]
Hello,
how can I use DateTime field (i can not use it in Preset). 
Or am I doing something wrong?
Thank you

**Comment by voidgraphics** [comment]
While I haven't tried, DateTime fields should work fine in Presets. The usage should be the same as other fields:

```php
namespace App\Nova\Flexible\Presets;

use Laravel\Nova\Fields\DateTime;
use Whitecube\NovaFlexibleContent\Flexible;
use Whitecube\NovaFlexibleContent\Layouts\Preset;

class MyPreset extends Preset
{

    /**
     * Execute the preset configuration
     *
     * @return void
     */
    public function handle(Flexible $field)
    {
        $field->button('Add an item');
        $field->addLayout('My layout', 'my-layout', [
            DateTime::make('My DateTime field'),
            // ...
        ]); 
    }
}
```

 Do you get an error message of some kind?

**Comment by Nyratas** [comment]
Hi @nejckorosec,

There seems to be an issue with date casting, indeed. I tried using a Datetime field inside a Custom Layout. Creating a new resource seems to work properly (the datetime value is stored in the JSON as expected) but displaying or updating the resource throws an error because the date attribute doesn't cast to a date.

Without casting, you should see the following error:
```
DateTime field must cast to 'datetime' in Eloquent model
```

When adding the [casts](https://github.com/laravel/nova-issues/issues/497) array to the Custom Layout as follows:
```php
    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [
        'my_date_field' => 'datetime'
    ];
```

It now throws:
```
Call to undefined method App\Nova\Flexible\Layouts\MyCustomLayoutTest::getConnection() 
```

Which means there is indeed an issue with date casting in Custom Layout classes.

Thank you for bringing it to our attention, we'll take a look at it shortly.

**Comment by sheggi** [comment]
HasAttributes expects `App\Nova\Flexible\Layouts\MyCustomLayoutTest` to be a Model.

When adding this function to the custom Layout, it will work
```
    public function getConnection()
    {
        return Model::resolveConnection(null);
    }
```

The problem with this solution is, it won't reflect the correct connection of the underlying model used by this field.

**Comment by GarethSomers** [comment]
As an intermediate resolution I've created custom Date field (Solution also applicable to DateTime).

```
<?php

namespace App\Nova\Fields;

use Carbon\Carbon;
use Laravel\Nova\Fields\Date as NovaDate;

class Date extends NovaDate
{
    /**
     * Create a new field.
     *
     * @param  string  $name
     * @param  string|null  $attribute
     * @param  mixed|null  $resolveCallback
     * @return void
     */
    public function __construct($name, $attribute = null, $resolveCallback = null)
    {
        parent::__construct($name, $attribute, function ($value) {
            if (is_string($value)) {
                try {
                    $possible_date = Carbon::createFromFormat('Y-m-d', $value);
                    if ($possible_date) {
                        $value = $possible_date;
                    }
                } catch(\Exception $e) {
                    return null;
                }
            }

            return $value->format('Y-m-d');
        });
    }
}
```

**Comment by Nyratas** [comment]
This issue was solved in a way that does not require to define a connection on the layouts. DateTime fields should now work as expected. 

Two side notes, though:

1. `DateTime` attributes should be defined in the `$casts` property: 
```php
    protected $casts = [
        'my_date' => 'datetime'
    ];
```
2. The default casting format is `Y-m-d H:i:s`. If your database connection has another storage format, you can specify it by defining a `$dateFormat`  property on the  layout: 
```php
    /**
     * The storage format of the model's date columns.
     *
     * @var string
     */
    protected $dateFormat = 'Y-m-d H:i:s.u';
```

This fix will be included in the next release, `v0.1.8`.
